rest/rest-client/src/main/java/org/ow2/proactive/scheduler/rest/SchedulerClient.java

```diff
@@ -988,26 +989,28 @@ public boolean deleteFile(String space, String pathname) throws NotConnectedExce
     @Override
     public void renewSession() throws NotConnectedException {
         Closer closer = Closer.create();
-        try {
-            LoginForm loginForm = new LoginForm();
-            loginForm.setUsername(connectionInfo.getLogin());
-            loginForm.setPassword(connectionInfo.getPassword());
-            if (connectionInfo.getCredentialFile() != null) {
-                FileInputStream inputStream = new FileInputStream(connectionInfo.getCredentialFile());
+
+        LoginForm loginForm = new LoginForm();
+        loginForm.setUsername(connectionInfo.getLogin());
+        loginForm.setPassword(connectionInfo.getPassword());
+
+        if (connectionInfo.getCredentialFile() != null) {
+            try (FileInputStream inputStream = new FileInputStream(connectionInfo.getCredentialFile())) {
+
                 closer.register(inputStream);
                 loginForm.setCredential(inputStream);
+
+            } catch (Exception e) {
+                throw new RuntimeException(e);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Define and throw a dedicated exception instead of using a generic one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00112)